src/directional_clustering/plotters/ply_plotter.py

```diff
@@ -0,0 +1,138 @@
+from numpy import zeros
+from numpy import asarray
+
+# plotter is based on Plotly (https://plotly.com/python/)
+import plotly.graph_objects as go
+import plotly.figure_factory as ff
+
+from compas.geometry import length_vector
+
+from directional_clustering.plotters import mesh_to_vertices_xyz
+from directional_clustering.plotters import lines_to_start_end_xyz
+from directional_clustering.plotters import lines_start_end_connect
+from directional_clustering.plotters import vectors_dict_to_array
+from directional_clustering.plotters import face_centroids
+from directional_clustering.plotters import line_sdl
+from directional_clustering.plotters import rgb_colors
+
+__all__ = [
+    "PlyPlotter"
+]
+
+# TODO: see decorator style
+
+class PlyPlotter(go.Figure):
+    """
+    PlyPlotter is a custom wrapper around a Plotly.py graph object plotter.
+    """
+
+    def __init__(self, *args):
+        super(PlyPlotter, self).__init__(*args)
+
+    def set_title(self, title):
+        """
+        Sets title of the plot and sets the aspect ratio to the data. The default aspect ratio is of a cube, which distorts the image.
+        """
+        self.update_layout(
+            title_text=title,
+            showlegend=False,
+            scene=dict(aspectmode='data'))
+
+    def draw_vector_field_array(self, mesh, vectors, color, uniform, scale, width=0.5):
+        """
+        Plots the vector field.
+        """
+        field = vectors_dict_to_array(vectors, mesh.number_of_faces())
+        lines = []
+
+        # get lines
+        rows, _ = field.shape
+        for fkey in range(rows):
+            vector = field[fkey]
+            vector = vector.tolist()
+
+            if uniform:
+                vec_length = scale
+            else:
+                vec_length = length_vector(vector) * scale
+
+            pt = mesh.face_centroid(fkey)
+            lines.append(line_sdl(pt, vector, vec_length))
+
+        # "s" is shorthand for start, "e" is shorthand for end
+        s_x, s_y, s_z, e_x, e_y, e_z = lines_to_start_end_xyz(lines)
+
+        # "cse" is shorthand for connected start and end
+        cse_x, cse_y, cse_z = lines_start_end_connect(s_x, s_y, s_z,
+            e_x, e_y, e_z)
+
+        # add lines to plot
+        self.add_trace(
+            go.Scatter3d(
+                x=cse_x,
+                y=cse_y,
+                z=cse_z,
+                mode='lines',
+                line=dict(width=2, color=color),
+                opacity=1
+            ))
+
+    def draw_trimesh(self, mesh, paint_clusters, draw_edges=False, opacity=0.8):
+        """
+        Plots the mesh with or without its edges, depending on what is chosen in `draw_edges`. Draws face colors according to values in each face.
```

Comment:
Suggestion: Breaking long one-liners into two lines (line length <80 characters as per PEP)

Reply:
Ah, this is annoying: I thought my automatic line breaker did this automatically, but apparently, I have to add an enter myself... thanks for pointing it out.